pkg/workers/worker_interface_test.go

```diff
@@ -0,0 +1,268 @@
+			b := &BaseWorker{
+				WorkerType: tt.fields.WorkerType,
+			}
+			g.Expect(b.GetWorkerType()).To(Equal(tt.want))
+		})
+	}
+}
+
+func TestBaseWorker_GetStopChan(t *testing.T) {
+	type fields struct {
+		imStop chan struct{}
+	}
+	var stopchan chan struct{}
+	tests := []struct {
+		name   string
+		fields fields
+		want   *chan struct{}
+	}{
+		{
+			name: "should return the base workers stop chan",
+			fields: fields{
+				imStop: stopchan,
+			},
+			want: &stopchan,
+		},
+	}
+	g := NewWithT(t)
+	for _, testcase := range tests {
+		tt := testcase
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			b := &BaseWorker{
+				imStop: tt.fields.imStop,
+			}
+			g.Expect(b.GetStopChan()).To(Equal(tt.want))
+		})
+	}
+}
+
+func TestBaseWorker_GetSyncGroup(t *testing.T) {
+	type fields struct {
+		syncTeardown *sync.WaitGroup
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		want   *sync.WaitGroup
+	}{
+		{
+			name: "should return the base workers sync group",
+			fields: fields{
+				syncTeardown: &sync.WaitGroup{},
+			},
+			want: &sync.WaitGroup{},
+		},
+	}
+	g := NewWithT(t)
+	for _, testcase := range tests {
+		tt := testcase
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			b := BaseWorker{
+				syncTeardown: sync.WaitGroup{},
+			}
+			g.Expect(b.GetSyncGroup()).To(Equal(tt.want))
+		})
+	}
+}
+
+func TestBaseWorker_IsRunning(t *testing.T) {
+	type fields struct {
+		isRunning bool
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		want   bool
+	}{
+		{
+			name: "should return true if the worker is running",
+			fields: fields{
+				isRunning: true,
+			},
+			want: true,
+		},
+	}
+	g := NewWithT(t)
+	for _, testcase := range tests {
+		tt := testcase
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			b := &BaseWorker{
+				isRunning: tt.fields.isRunning,
+			}
+			g.Expect(b.IsRunning()).To(Equal(tt.want))
+		})
+	}
+}
+
+func TestBaseWorker_SetIsRunning(t *testing.T) {
+	type fields struct {
+		isRunning bool
+	}
+	type args struct {
+		val bool
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		args   args
+		want   bool
+	}{
+		{
+			name: "should set the variable isRunning",
+			fields: fields{
+				isRunning: false,
+			},
+			args: args{
+				val: true,
+			},
+			want: true,
+		},
+	}
+	g := NewWithT(t)
+	for _, testcase := range tests {
+		tt := testcase
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			b := &BaseWorker{
+				isRunning: tt.fields.isRunning,
+			}
+			b.SetIsRunning(tt.args.val)
+			g.Expect(b.IsRunning()).To(Equal(tt.want))
+		})
+	}
+}
+
+func TestBaseWorker_StartWorker(t *testing.T) {
+	type fields struct {
+		Id         string
+		WorkerType string
+		Reconciler Reconciler
+		isRunning  bool
+		imStop     chan struct{}
+	}
+	var stopchan chan struct{}
+	var wg sync.WaitGroup
+	type args struct {
+		w Worker
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		args   args
+	}{
+		{
+			name: "should start then stop the worker",
```

Comment:
Why start and then stop the worker? Shouldn't this be testing start and stop methods separately?

Reply:
+1, let's separate them.